libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineColorResources.kt

```diff
@@ -58,7 +60,8 @@ class RouteLineColorResources private constructor(
     @ColorInt val restrictedRoadColor: Int,
     @ColorInt val alternativeRouteRestrictedRoadColor: Int,
     @ColorInt val routeClosureColor: Int,
-    @ColorInt val alternativeRouteClosureColor: Int
+    @ColorInt val alternativeRouteClosureColor: Int,
+    @ColorInt val deEmphasizeInActiveRouteLegsColor: Int,
```

Comment:
As mentioned above, de-emphasizing is only one of the use cases. We should probably expose all available styling attributes equally for the active and inactive legs.

 What if we refactor color resources into:
```kotlin
class ColorResources private constructor(
    @ColorInt val color: Int,
    @ColorInt val casingColor: Int,
    @ColorInt val unknownCongestionColor: Int,
    @ColorInt val lowCongestionColor: Int,
    @ColorInt val moderateCongestionColor: Int,
    @ColorInt val heavyCongestionColor: Int,
    @ColorInt val severeCongestionColor: Int,
    @ColorInt val restrictedRoadColor: Int,
    @ColorInt val routeClosureColor: Int,
)

class RouteLineColorResources private constructor(
    @ColorInt val routeLineTraveledColor: Int,
    @ColorInt val routeLineTraveledCasingColor: Int,
    val primaryRouteColorResources: ColorResources,
    val alternativeRouteColorResources: ColorResources,
    val inactivePrimaryRouteLegsColorResources: ColorResources
)
```

This way we ensure that we support all of the possibilities equally, and we clean up the repetitions in the API.

Reply:
There's only one color for the inactive route legs. I changed the name.

Reply:
What do you think about the proposal in general, clean up the API?

Reply:
Personally I don't see an advantage. The only colors in common are the base colors and traffic colors. The other colors don't apply to alternative routes. In addition these are already 3 levels deep `MapboxRouteLineOptions->RouteLineResources->RouteLineColorResources`. Your suggestion would make many of the colors 4 levels deep. My preference is to leave them as is.